pkg/table/tables/tables.go

```diff
@@ -1974,29 +1973,23 @@ func FindIndexByColName(t table.Table, name string) table.Index {
 	return nil
 }
 
-func getDuplicateErrorHandleString(t table.Table, handle kv.Handle, row []types.Datum) string {
+func getDuplicateErrorHandleString(tblInfo *model.TableInfo, handle kv.Handle, row []types.Datum) string {
 	if handle.IsInt() {
 		return kv.GetDuplicateErrorHandleString(handle)
 	}
-	var pk table.Index
-	for _, idx := range t.Indices() {
-		if idx.Meta().Primary {
-			pk = idx
-			break
-		}
+	// Here pkIdx must not be null because of hanle.IsInt() is false?
```

Comment:
Yes, but if the meta data is wrong, we don't want non-critical code like this to panic.

```
intest.AssertNotNil(pkIdx)
if pkIdx == nil {
    return kv.GetDuplicateErrorHandleString(handle)
}
```

Reply:
Updated.